src/plus/gk/account/subscriptionService.ts

```diff
@@ -488,6 +491,37 @@ export class SubscriptionService implements Disposable {
 		});
 	}
 
+	private requestExceptionCount = 0;
+	private requestsAreBlocked = false;
+
+	resetRequestExceptionCount(): void {
+		this.requestExceptionCount = 0;
+		this.requestsAreBlocked = false;
+	}
+
+	trackRequestException(): void {
+		this.requestExceptionCount++;
+
+		if (this.requestExceptionCount >= 5 && this.getAuthenticationSession() !== null) {
```

Comment:
Why do we need to check that there is an active session here?

Would the following simplified version work?
```
if (this.requestExceptionCount >= 5 && !this.requestsAreBlocked) {
   void showIntegrationDisconnectedTooManyFailedRequestsWarningMessage('GitKraken');
   this.requestsAreBlocked = true;
   this.requestExceptionCount = 0;
}
```

Reply:
fixed